platform/SharedPtr.h

```diff
@@ -75,6 +75,13 @@ class SharedPtr {
     {
     }
 
+    /**
+     * @brief Create empty SharedPtr not pointing to anything.
+     */
+    constexpr SharedPtr(std::nullptr_t) : SharedPtr()
```

Comment:
You could use constexpr to the default constructor as well.

Reply:
Indeed.

I'm doing that in #12037 - I've split the PRs cos this one is a breaking change, and that one isn't.

Reply:
And, actually, the `constexpr` here is ineffective until the default constructor it delegates to is made `constexpr`. But it's harmless if added alone.